### PR TITLE
CTCTRADERS-532:Show user the MRN page and allow user to correct

### PR DESCRIPTION
**Pull request by ashwinirudrappa**

### CODE REVIEW

File: app/connectors/ArrivalMovementConnector.scala [hunk 1]
```diff
@@ -57,15 +57,19 @@ class ArrivalMovementConnector @Inject()(val config: FrontendAppConfig, val http
...
     val serviceUrl = s"${config.baseDestinationUrl}$location"
     http.GET[HttpResponse](serviceUrl) map {
       case responseMessage if is2xx(responseMessage.status) =>
-        Some(responseMessage.json.as[ResponseMovementMessage].message)
+        val xml = responseMessage.json.as[ResponseMovementMessage].message
+
+        MovementReferenceNumber(xml.\\("DocNumHEA5").text) map {
+          mrn =>
+            (responseMessage.json.as[ResponseMovementMessage].message, mrn)
+        }
```
Comment (amaalali) [inline app/connectors/ArrivalMovementConnector.scala:70]: This feels like a separate concern that the Connector shouldn't be dealing with

File: app/controllers/UpdateRejectedMRNController.scala [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+          formWithErrors => {
+
+            val json = Json.obj("form" -> formWithErrors)
+
+            renderer.render("movementReferenceNumber.njk", json).map(BadRequest(_))
+          },
+          value => Future(Redirect(navigator.nextRejectionPage(UpdateRejectedMovementReferenceNumberPage, value, arrivalId)))
+        )
+  }
+}
```
Comment (amaalali) [inline app/controllers/UpdateRejectedMRNController.scala:72]: Interesting, why not reuse the existing controller? I'm not suggesting that we do, it's just that reusing this microservice for the IE007 fix declaration journey is meant to reuse components from the IE007 new declaration journey, so are we getting any benefit from bloating this microservice?

File: app/navigation/Navigator.scala [hunk 3]
```diff
@@ -108,6 +108,12 @@ class Navigator @Inject()() {
       }
   }
 
+  def nextRejectionPage(page: Page, mrn: MovementReferenceNumber, arrivalId: ArrivalId): Call = 
+    page match {
+      case UpdateRejectedMovementReferenceNumberPage => routes.CheckYourAnswersRejectionsController.onPageLoad(mrn, arrivalId)
+      case _ => routes.TechnicalDifficultiesController.onPageLoad()
+    }
```
Comment (amaalali) [inline app/navigation/Navigator.scala:115]: Why create a new interface on this as opposed to a new mode type?

File: app/services/ArrivalNotificationMessageService.scala [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+class ArrivalNotificationMessageService @Inject()(arrivalMovementConnector: ArrivalMovementConnector) {
+
+  def getArrivalNotificationMessage(arrivalId: ArrivalId)(implicit hc: HeaderCarrier,
+                                                          ec: ExecutionContext): Future[Option[(NodeSeq, MovementReferenceNumber)]] =
+    arrivalMovementConnector.getSummary(arrivalId) flatMap {
+      case Some(summary) =>
+        arrivalMovementConnector.getArrivalNotificationMessage(summary.messagesLocation.arrivalNotification)
+      case _ => Future.successful(None)
+    }
```
Comment (amaalali) [inline app/services/ArrivalNotificationMessageService.scala:35]: This feels like the layer that should be parsing the data from the API call

File: app/viewModels/ErrorTypeViewModel.scala [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package viewModels
+
+class ErrorTypeViewModel {}
```
Comment (amaalali) [inline app/viewModels/ErrorTypeViewModel.scala:19]: Is there future plans for this?

File: test/services/ArrivalNotificationMessageServiceSpec.scala [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+  override def beforeEach: Unit = {
+    super.beforeEach
+    reset(mockConnector)
+  }
+
+  private val arrivalId = ArrivalId(1)
+
+  "ArrivalNotificationMessageService" - {
+    "must return NodeSeq for the input arrivalId" in {
```
Comment (amaalali) [inline test/services/ArrivalNotificationMessageServiceSpec.scala:42]: The description isn't quite accurate

File: app/controllers/UpdateRejectedMRNController.scala [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+                                            renderer: Renderer)(implicit ec: ExecutionContext)
+    extends FrontendBaseController
+    with I18nSupport
+    with NunjucksSupport {
+
+  private val form = formProvider()
+
+  def onPageLoad(arrivalId: ArrivalId): Action[AnyContent] = identify.async {
+    implicit request =>
+      arrivalMovementMessageService.getArrivalNotificationMessage(arrivalId) flatMap {
```
Comment (amaalali) [inline app/controllers/UpdateRejectedMRNController.scala:50]: Shouldn't the data already be loaded by the time the user is here? We do an initial load when the user is on the error list page

File: app/connectors/ArrivalMovementConnector.scala (hunk 1 above)
Comment (ashwinirudrappa) [reply app/connectors/ArrivalMovementConnector.scala:70]: @amaalali Yes, It was a fix for now that's the reason we placed a TODO

File: app/controllers/UpdateRejectedMRNController.scala (hunk 7 above)
Comment (amaalali) [reply app/controllers/UpdateRejectedMRNController.scala:50]: As discussed with the team caching to be done as part of future PR to follow

**Pull request merged**